src/dispatch/workflow/models.py

```diff
@@ -0,0 +1,169 @@
+from datetime import datetime
+from enum import Enum
+from typing import List, Optional
+
+from pydantic import validator
+from sqlalchemy.orm import relationship, backref
+from sqlalchemy import Column, ForeignKey, Integer, String, JSON, Table
+from sqlalchemy.sql.schema import PrimaryKeyConstraint
+from sqlalchemy.sql.sqltypes import Boolean
+from sqlalchemy_utils import TSVectorType
+
+from dispatch.database import Base
+from dispatch.document.models import DocumentCreate
+from dispatch.models import DispatchBase, ResourceMixin, TimeStampMixin
+from dispatch.participant.models import ParticipantRead
+from dispatch.plugin.models import PluginRead
+
+
+class WorkflowInstanceStatus(str, Enum):
+    submitted = "submitted"
+    created = "created"
+    running = "running"
+    completed = "completed"
+    failed = "failed"
+
+
+# Association tables for many to many relationships
+assoc_workflow_instances_artifacts = Table(
+    "workflow_instance_artifact",
+    Base.metadata,
+    Column("document_id", Integer, ForeignKey("document.id")),
+    Column("workflow_instance_id", Integer, ForeignKey("workflow_instance.id")),
+    PrimaryKeyConstraint("document_id", "workflow_instance_id"),
+)
+
+assoc_workflow_incident_priorities = Table(
+    "workflow_incident_priority",
+    Base.metadata,
+    Column("incident_priority_id", Integer, ForeignKey("incident_priority.id")),
+    Column("workflow_id", Integer, ForeignKey("workflow.id")),
+    PrimaryKeyConstraint("incident_priority_id", "workflow_id"),
+)
+
+assoc_workflow_incident_types = Table(
+    "workflow_incident_type",
+    Base.metadata,
+    Column("incident_type_id", Integer, ForeignKey("incident_type.id")),
+    Column("workflow_id", Integer, ForeignKey("workflow.id")),
+    PrimaryKeyConstraint("incident_type_id", "workflow_id"),
+)
+
+assoc_workflow_terms = Table(
+    "workflow_term",
+    Base.metadata,
+    Column("term_id", Integer, ForeignKey("term.id")),
+    Column("workflow_id", Integer, ForeignKey("workflow.id")),
+    PrimaryKeyConstraint("term_id", "workflow_id"),
+)
+
+
+class Workflow(Base, TimeStampMixin):
```

Comment:
Should we add a column for the creator of the flow, so people know who to reach out to if they have questions or feedback?

Reply:
Perhaps, I think we can add this in later once we have workflow owners that aren't SIRT.